langchain/vectorstores/redisearch.py

```diff
@@ -0,0 +1,189 @@
+        )
+        redis_query = (
+            Query(base_query)
+            .return_fields(*return_fields)
+            .sort_by("vector_score")
+            .paging(0, k)
+            .dialect(2)
+        )
+        params_dict: Mapping[str, str] = {
+            "vector": str(np.array(embedding).astype(dtype=np.float32).tobytes())
+        }
+
+        # perform vector search
+        results = self.client.ft(self.index_name).search(redis_query, params_dict)
+
+        documents = [
+            Document(page_content=result.content, metadata=json.loads(result.metadata))
+            for result in results.docs
+        ]
+
+        return documents
+
+    @classmethod
+    def from_texts(
+        cls,
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        index_name: Optional[str] = None,
+        **kwargs: Any,
+    ) -> RediSearch:
+        """Construct RediSearch wrapper from raw documents.
+        This is a user-friendly interface that:
+            1. Embeds documents.
+            2. Creates a new index for the embeddings in the RediSearch instance.
+            3. Adds the documents to the newly created RediSearch index.
+        This is intended to be a quick way to get started.
+        Example:
+            .. code-block:: python
+                from langchain import RediSearch
+                from langchain.embeddings import OpenAIEmbeddings
+                embeddings = OpenAIEmbeddings()
+                redisearch = RediSearch.from_texts(
+                    texts,
+                    embeddings,
+                    redisearch_url="redis://username:password@localhost:6379"
+                )
+        """
+        redisearch_url = get_from_dict_or_env(
+            kwargs, "redisearch_url", "REDISEARCH_URL"
+        )
+        try:
+            import redis
+            from redis.commands.search.field import TextField, VectorField
+            from redis.commands.search.indexDefinition import IndexDefinition, IndexType
+        except ImportError:
+            raise ValueError(
+                "Could not import redis python package. "
+                "Please install it with `pip install redis`."
+            )
+        try:
+            pool = redis.ConnectionPool.from_url(redisearch_url)
+            client = redis.StrictRedis(connection_pool=pool)
+        except ValueError as e:
+            raise ValueError(f"Your redis connected error: {e}")
+        embeddings = embedding.embed_documents(texts)
+        dim = len(embeddings[0])
+        # Constants
+        vector_number = len(embeddings)  # initial number of vectors
+        # name of the search index if not given
+        if not index_name:
+            index_name = uuid.uuid4().hex
+        prefix = "doc"  # prefix for the document keys
+        distance_metric = (
+            "COSINE"  # distance metric for the vectors (ex. COSINE, IP, L2)
```

Comment:
Allow this to be changed?

Reply:
This will work for now, but for many production use cases - users will need the ability to define and specify index name, prefix, and distance metrics. But that can come later as optional arguments.

Reply:
@tylerhutcherson expressed the main point behind what I meant. users will want to use HNSW for larger indices and change distance metrics based on process and normalization. but for now, this is fine and awesome to see.